src/averageAges.js

```diff
@@ -56,6 +77,25 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const peopleWithAgeDiff = people.map((person, i, array) => {
+    const momIndex = array.findIndex(a => a.name === person.mother);
+
+    if (momIndex !== -1) {
+      person.ageDiff = person.born - array[momIndex].born;
+    }
+
+    return person;
+  }).filter(a => a.ageDiff);
+
+  const childWithAgeDiff = onlyWithSon
+    ? peopleWithAgeDiff.filter(a => a.sex === 'm')
+    : peopleWithAgeDiff;
```

Comment:
Use filter instead of map to get needed objects, and then calculate age difference in `reduce` method